### PR TITLE
FreeBSD rc.d service script update

### PR DESCRIPTION
**Pull request by n-connect**
Updating the files, contained authelia.service, to have the fresh authelia-fb-rc.d as well, inculding the  .buildkite/hooks/pre-artifact with and if elif else condition

## Summary by CodeRabbit

- **Chores**
	- Introduced a new script for managing the `authelia` daemon service, enabling easier configuration and control.
	- Updated `.dockerignore` to include `authelia-fb-rc.d` for improved Docker build efficiency.
	- Enhanced the `.buildkite/hooks/pre-artifact` script to handle different operating systems when creating tar archives, improving cross-platform compatibility and notifying when an unsupported OS is encountered.
	- Added documentation for publishing an `rc.d` service script file for Authelia on FreeBSD.

### CODE REVIEW

**Comment by n-connect** [comment]
Hello all,

It seems the release files at [buildkite/authelia](https://buildkite.com/authelia/authelia/builds/27936) created, all with their proper service script file. Let me know if you need something else.

File: docs/content/integration/deployment/bare-metal.md [hunk 1]
```diff
@@ -62,6 +62,10 @@ $ nix-env -iA nixpkgs.authelia
 In addition to the [binaries](#binaries) we publish, [FreshPorts](https://www.freshports.org/www/authelia/) offer a
 package.
 
+We publish an [rc.d](https://people.freebsd.org/~olivierd/porters-handbook/rc-scripts.html) service script file:
```
Comment (decke) [inline docs/content/integration/deployment/bare-metal.md:65]: That link is really internal. You probably want to use: https://docs.freebsd.org/en/articles/rc-scripting/

File: authelia-fb-rc.d [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+load_rc_config $name
+
+authelia_enable=${authelia_enable:-"NO"}
+
+logfile="/var/log/${name}.log"
+
+procname=/usr/local/bin/authelia
+command="/usr/sbin/daemon"
+command_args="-u root -o ${logfile} -t ${name} /usr/local/bin/authelia --config /usr/local/etc/authelia.yml"
```
[the remark this answers is not in the document]
Comment (decke) [reply authelia-fb-rc.d:23]: For better or worse this is a copy of the rc.d script from the FreeBSD port at https://cgit.freebsd.org/ports/tree/www/authelia/files/authelia.in

**Review by decke — COMMENTED** [review]
It is really not clear to me what kind of benefit this PR provides and especially for which use case.

As a FreeBSD Ports committer I can only say this does not help us in the FreeBSD Ports context.

**Comment by n-connect** [comment]
@decke 

> As a FreeBSD Ports committer I can only say this does not help us in the FreeBSD Ports context.

This PR has no intention to do anything with FreeBSD Ports at all, just fixing the error of packaging a Linux SystemD service script with a FreeBSD binary build.  Yes, the script is a identical copy of the original rc.d fle from the FreeBSD pkg release.

> It is really not clear to me what kind of benefit this PR provides and especially for which use case.

Use case and benefit is being able to download directly and use the latest Authelia directly from github, right when it is available. Authelia packs big changes in its releases

File: docs/content/integration/deployment/bare-metal.md (hunk 1 above)
Comment (n-connect) [reply docs/content/integration/deployment/bare-metal.md:65]: No problem, can be replaced.

File: .buildkite/hooks/pre-artifact [hunk 3]
```diff
@@ -11,7 +11,14 @@ if [[ "${BUILDKITE_LABEL}" == ":hammer_and_wrench: Unit Test" ]]; then
     for OS in "${!BUILDS[@]}"; do
       for ARCH in ${BUILDS[$OS]}; do
         tar -czf "authelia-${OS}-${ARCH}.tar.gz" "authelia-${OS}-${ARCH}" authelia.service config.template.yml
-        sha256sum "authelia-${OS}-${ARCH}.tar.gz" > "authelia-${OS}-${ARCH}.tar.gz.sha256"
+         if [[ ${OS} == linux ]]; then
+          tar -czf "authelia-${OS}-${ARCH}.tar.gz" "authelia-${OS}-${ARCH}" authelia.service config.template.yml
```
[the remark this answers is not in the document]
Comment (n-connect) [reply .buildkite/hooks/pre-artifact]: Fixed

**Review by james-d-elliott — APPROVED** [review]
LGTM

**Pull request merged**